proposals/readonly-setter-calls-on-non-variables.md

```diff
@@ -77,7 +77,27 @@ These use cases would no longer be blocked if CS1612 is fully updated with an un
 
 The CS1612 error is not produced for assignments where the setter is readonly. (If the whole struct is readonly, then the setter is also readonly.) The setter call is emitted the same way as any non-accessor readonly instance method call.
 
-### Notes
+### Null-conditional assignment
+
+The [Null-conditional assignment](https://github.com/dotnet/csharplang/blob/main/proposals/null-conditional-assignment.md) proposal enables the following syntax:
+
+```cs
+a?.b = value;
+a?.b.c = value;
+a?[index] = value;
+```
+
+CS1612 errors will be produced for this new syntax in the same way that they are for the corresponding forms when `a` is non-nullable:
```

Comment:
Why do they need to be restricted to non-nullable? If it was a nullable the same logic for allowing the `set` applies. It's just writing through to heap state.

Reply:
I didn't intend this as a restriction, merely "here's the _existing_ syntax that parallels the new syntax mentioned above."